src/natcap/invest/urban_nature_access.py

```diff
@@ -547,6 +655,376 @@ def execute(args):
     LOGGER.info('Finished Urban Nature Access Model')
 
 
+def _find_disjoint_polygon_groups(aoi_vector_path):
+    """Identify truly disjoint polygon groups.
+
+    Pygeoprocessing's calculate_disjoint_polygon_set() uses any form of
+    intersection (even if polygons touch but do not overlap) in determining
+    whether polygons are disjoint.  When a user provides a vector of AOIs that
+    touch but do NOT overlap, we do not need to use pygeoprocessing's set of
+    disjoint polygons, which makes more sense visually.
+
+    Args:
+        aoi_vector_path (string): A path to the AOI vector on disk.
+
+    Returns:
+        A list of lists of disjoint FIDs.
+    """
+    all_geometries = []
+    vector = ogr.Open(aoi_vector_path)
+    layer = vector.GetLayer()
+    area_sum = 0
+    all_fids = []
+    for feature in layer:
+        geom = feature.GetGeometryRef()
+        all_geometries.append(shapely.wkt.loads(geom.ExportToWkt()))
+        area_sum += geom.Area()
+        all_fids.append(feature.GetFID())
+
+    union_geom = shapely.ops.unary_union(all_geometries)
+    union_area = union_geom.area
+    if math.isclose(union_area, area_sum):
+        return [all_fids]
+
+    LOGGER.warning(
+        f"Some admin units overlap by {abs(union_area - area_sum)} square "
+        "units. Falling back to disjoint polygon sets."
+    )
+    return calculate_disjoint_polygon_set(
+        aoi_vector_path, ok_for_geometries_to_touch=True)
+
+
+def calculate_disjoint_polygon_set(
+        vector_path, layer_id=0, bounding_box=None,
+        ok_for_geometries_to_touch=False):
+    """Create a sequence of sets of polygons that don't overlap.
+
+    Determining the minimal number of those sets is an np-complete problem so
+    this is an approximation that builds up sets of maximal subsets.
```

Comment:
Smart.

Reply:
Ha, I really can't claim credit for this ... this is Rich's handiwork: https://github.com/natcap/pygeoprocessing/blob/main/src/pygeoprocessing/geoprocessing.py#L2232

it _is_ a cool algorithm, though, and very handy!  I copied this function in here so that I could modify the logic determining which features are disjoint since we have an interesting case where we expect our admin geometries to be nonoverlapping.